src/main/java/com/mindsdb/kafka/connect/client/MindsDBCloudClientImpl.java

```diff
@@ -0,0 +1,109 @@
+package com.mindsdb.kafka.connect.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.mindsdb.kafka.connect.client.models.Predictor;
+import org.apache.kafka.connect.errors.ConnectException;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class MindsDBCloudClientImpl implements MindsDBClient {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    public static final String CONNECTION_ERROR = "Failed to send request to MindsDB server";
+
+    private final HttpClient httpClient;
+    private final MindsDBClientConfig clientConfig;
+
+    public MindsDBCloudClientImpl(HttpClient httpClient, MindsDBClientConfig clientConfig) {
+        this.httpClient = httpClient;
+        this.clientConfig = clientConfig;
+
+        try {
+            postToMindsDb(
+                    "/cloud/login",
+                    this.clientConfig.cloudLoginBody()
+            );
+        } catch (Exception e) {
+            throw new ConnectException("Failed to authenticate to cloud with the provided user/password", e);
+        }
+    }
+
+    @Override
+    public void createIntegration() { }
+
+    @Override
+    public void createStream() throws MindsDBApiException {
+        postToMindsDb(
+                clientConfig.streamCreationUri(),
+                clientConfig.cloudStreamRequest()
+        );
```

Comment:
Need to handle "`session` cookie is expired" case. What if:
1. catch MindsDBApiException
2. call `login` method in the `catch` body
3. refresh `session` cookie value in success, throws MindsDBApiException which wouldn't be handled otherwise.
4. call `postToMindsDb` with the same params again